hcloud/client.go

```diff
@@ -410,6 +414,12 @@ func (r *Response) readMeta(body []byte) error {
 	return nil
 }
 
+// InternalCorrelationID returns the unique ID of the request as set by the API. This ID can help with support requests,
+// as it allows the people working on identify this request in particular.
+func (r *Response) InternalCorrelationID() string {
+	return r.Header.Get(headerCorrelationID)
+}
+
```

Comment:
I am not sure if we should add this to our public interface. There is no guarantee that this header will always be exposed in the API. I am fine with removing this from the public interface.

Reply:
I'd keep it private for now, until we have a better vision on how we want to do tracing in our integrations.

I am not sure if we have plans for extending the reach of our tracing capabilities, but I think it would be awesome to not stop tracing at the boundaries of the public API, but try to push it all the way in our integrations' logic.

Maybe we are allowed to generate this correlation ID, and let the API reuse it or wrap it. I'd love to have the traces that start from a hccm call and come all the way back :).

Reply:
Changed to `internalCorrelationID` (private)